### PR TITLE
New/Set-GitHubRepository: Add Support for 'Allow Update Branch' Option

### LINKED ISSUE #386: New/Set-GitHubRepository: Add Support for 'Allow Update Branch' Option
#### Feature Idea Summary

GitHub have added a new option to always have the option to update a pull request branch for a repository. This should be added to the `New-GitHubRepository` and `Set-GitHubRepository` Cmdlets.

##### References:

- [More ways to keep your pull request branch up-to-date](https://github.blog/changelog/2022-02-03-more-ways-to-keep-your-pull-request-branch-up-to-date/)
- [Managing suggestions to update pull request branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-suggestions-to-update-pull-request-branches)

##### API Reference

- https://docs.github.com/en/rest/repos/repos#create-an-organization-repository
- https://docs.github.com/en/rest/repos/repos#update-a-repository

##### Parameters

- `allow_update_branch`

#### Requested Assignment

- If possible, I would like to implement this.

### PR DESCRIPTION
**Pull request by X-Guardian**
#### Description

This PR adds the `AllowUpdateBranch` parameter to the `New-GitHubRepository` and `Set-GitHubRepository` functions:

#### Issues Fixed

- Fixes #386

#### References

- https://docs.github.com/en/rest/repos/repos#create-an-organization-repository
- https://docs.github.com/en/rest/repos/repos#update-a-repository

#### Checklist

- [x] You actually ran the code that you just wrote, especially if you did just "one last quick change".
- [x] Comment-based help added/updated, including examples.
- [x] [Static analysis](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#static-analysis) is reporting back clean.
- [x] New/changed code adheres to our [coding guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#coding-guidelines).
- [ ] [Formatters were created](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#formatters) for any new types being added.
- [x] New/changed code continues to [support the pipeline](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#pipeline-support).
- [ ] Changes to the manifest file follow the [manifest guidance](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#module-manifest).
- [x] Unit tests were added/updated and are all passing. See [testing guidelines](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#testing).  This includes making sure that all pipeline input variations have been covered.
- [ ] Relevant usage examples have been added/updated in [USAGE.md](https://github.com/microsoft/PowerShellForGitHub/blob/master/USAGE.md).
- [ ] If desired, ensure your name is added to our [Contributors list](https://github.com/microsoft/PowerShellForGitHub/blob/master/CONTRIBUTING.md#contributors)

### CODE REVIEW

**Comment by X-Guardian** [comment]
@HowardWolosky, this PR is ready for review.

File: GitHubRepositories.ps1 [hunk 1]
```diff
@@ -89,6 +89,10 @@ filter New-GitHubRepository
     .PARAMETER AllowAutoMerge
         Specifies whether to allow auto-merge on pull requests.
 
+    .PARAMETER AllowUpdateBranch
+        Specifies whether to always allow a pull request head branch that is behind its base branch
+        to be updated even if it is not required to be up to date before merging.
```
Comment (HowardWolosky) [inline GitHubRepositories.ps1:94]:
```suggestion
        to be updated even if it is not required to be up-to-date before merging.
```

This is consistent with the API docs, but the docs are incorrect.

File: GitHubRepositories.ps1 [hunk 2]
```diff
@@ -1075,6 +1082,10 @@ filter Set-GitHubRepository
...
```
Comment (HowardWolosky) [inline GitHubRepositories.ps1:1087]:
```suggestion
        to be updated even if it is not required to be up-to-date before merging.
```

**Review by HowardWolosky — DISMISSED** [review]
Looks good, thanks for the contribution!

**Review by HowardWolosky — APPROVED** [review]

**Pull request merged**